cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -207,6 +207,7 @@ set_compiler_property(PROPERTY no_track_macro_expansion -ftrack-macro-expansion=
 
 # GCC 11 by default emits DWARF version 5 which cannot be parsed by
 # pyelftools. Can be removed once pyelftools supports v5.
+# pyelftools 0.31 still is not able to parse the version 5 output.
```

Comment:
I would prefer not to put a version number in here, cause chances are too low that it will not be updated when new pyelftools are released.

Instead an issue should be linked describing exactly what is needed, and then we can point to this issue instead.
Perhaps https://github.com/eliben/pyelftools/issues/581 covers the issue, else a new issue can be created.

pyelftools error seen:
```
  File ".../zephyr/scripts/logging/dictionary/database_gen.py", line 369, in extract_string_variables
    loc = loc_parser.parse_from_attribute(loc_attr, die.cu['version'])
  File ".../lib/python3.10/site-packages/elftools/dwarf/locationlists.py", line 309, in parse_from_attribute
    return self.location_lists.get_location_list_at_offset(
  File ".../lib/python3.10/site-packages/elftools/dwarf/locationlists.py", line 96, in get_location_list_at_offset
    raise DWARFError("For this binary, \"die\" needs to be provided")              
elftools.common.exceptions.DWARFError: For this binary, "die" needs to be provided
```

Side-note: initial and limited dwarf-5 support was included in 0.28, https://github.com/eliben/pyelftools/pull/363

Reply:
Yes, I saw this support for v5 and as I am already using 0.31 I thought this might work now. Obviously not.

To avoid some duplicated work some documentation of this fact might be nice. But I do not mind if we document the version or an actual issue. I'll try to figure out which issue we should reference ideally and will update this comment.

Reply:
I fully understand the frustration.
The original comment was made when there was no support for dwarf-5, while later releases do support dwarf-5 to some extent, then it seems Zephyr is still relying on parts not yet supported :disappointed: 

And thus it make sense to know what to track for future updates.

Thanks for looking into this :+1: